src/radix_connect/well_known_client/client.rs

```diff
@@ -0,0 +1,141 @@
+use crate::prelude::*;
+
+const SUFFIX_WELL_KNOWN_FILE: &str = ".well-known/dapp.json";
+
+pub struct WellKnownClient {
+    pub http_client: HttpClient,
+}
+
+impl WellKnownClient {
+    pub fn new(http_client: HttpClient) -> Self {
+        Self { http_client }
+    }
+
+    pub fn new_with_network_antenna(
+        network_antenna: Arc<dyn NetworkAntenna>,
+    ) -> Self {
+        Self::new(HttpClient::new(network_antenna))
+    }
+}
+
+impl NetworkRequest {
+    fn get_well_known_request(url: Url) -> Self {
+        let well_known_url = url.join(SUFFIX_WELL_KNOWN_FILE).unwrap();
+        NetworkRequest::new_get(well_known_url)
+    }
+}
+
+impl WellKnownClient {
+    pub async fn get_well_known_file(
+        &self,
+        url: Url,
+    ) -> Result<DappDefinitions> {
+        let network_request = NetworkRequest::get_well_known_request(url);
+        self.http_client
+            .execute_request_with_decoding(network_request)
+            .await
+    }
+}
+
+#[cfg(test)]
+impl WellKnownClient {
+    fn new_always_failing() -> Self {
+        Self::new_with_network_antenna(Arc::new(
+            MockAntenna::new_always_failing(),
+        ))
+    }
+
+    fn new_succeeding_http_client(response: &DappDefinitions) -> Self {
+        Self::new_with_network_antenna(Arc::new(MockAntenna::with_response(
+            response,
+        )))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_rt::time::timeout;
+    use std::time::Duration;
+    const MAX: Duration = Duration::from_millis(10);
+
+    const TEST_ORIGIN: &str = "https://d2xmq49o1iddud.cloudfront.net";
+
+    #[test]
+    fn test_get_well_known_request_error() {
+        let url = Url::parse(TEST_ORIGIN).unwrap();
+        let request = NetworkRequest::get_well_known_request(url.clone());
+        assert_eq!(request.url, url.join(SUFFIX_WELL_KNOWN_FILE).unwrap());
+        assert_eq!(request.method, NetworkMethod::Get);
+    }
+
+    #[actix_rt::test]
+    async fn test_get_well_known_file_correct_request_made() {
+        let mock_antenna_with_spy =
+            MockAntenna::with_spy(200, vec![], |request| {
+                let request = NetworkRequest::get_well_known_request(
+                    Url::from_str(TEST_ORIGIN).unwrap(),
+                );
+
+                let expected_request = NetworkRequest {
```

Comment:
prefer using a (or if lacking, writing a) constructor for `NetworkRequest`.

Reply:
and if you create expected `NetworkRequest` you can use `assert_eq!(request, expected)`;